console_win.go

```diff
@@ -496,79 +513,95 @@ func mrec2btns(mbtns, flags uint32) ButtonMask {
 }
 
 func (s *cScreen) getConsoleInput() error {
-	rec := &inputRecord{}
-	var nrec int32
-	rv, _, er := procReadConsoleInput.Call(
-		uintptr(s.in),
-		uintptr(unsafe.Pointer(rec)),
-		uintptr(1),
-		uintptr(unsafe.Pointer(&nrec)))
-	if rv == 0 {
-		return er
-	}
-	if nrec != 1 {
-		return nil
-	}
-	switch rec.typ {
-	case keyEvent:
-		krec := &keyRecord{}
-		krec.isdown = geti32(rec.data[0:])
-		krec.repeat = getu16(rec.data[4:])
-		krec.kcode = getu16(rec.data[6:])
-		krec.scode = getu16(rec.data[8:])
-		krec.ch = getu16(rec.data[10:])
-		krec.mod = getu32(rec.data[12:])
-
-		if krec.isdown == 0 || krec.repeat < 1 {
-			// its a key release event, ignore it
+	waitObjects := []syscall.Handle{s.in, s.cancelflag}
+
+	rv, _, er := procWaitForMultipleObjects.Call(
+		uintptr(len(waitObjects)),
+		uintptr(unsafe.Pointer(&waitObjects[0])),
```

Comment:
This logic is kind of subtle.  To be honest I'd really prefer to have some comment here.  We're doing some casting to assume addresses of arrays -- this is normal in C, but looks really unusual in Go, and Go programmers  may find this hard to grok.  I guess a comment like this:

// The address of the first element in the array is also the address of the array.

I'm not sure if that helps a lot, but it might.